### ISSUE TITLE
findAll() ignoring base attribute from @Entry

### ISSUE THREAD

**Issue by jenslofgren** [spring-projects/spring-ldap#31]
SimpleLdaprepository's findAll() wich in turn calls
```java
ldapOperations.findAll(clazz);
```

Seems to ignore any base set in the @Entry class

```java
    /**
     * {@inheritDoc}
     */
    @Override
    public <T> List<T> findAll(Class<T> clazz) {
        return findAll(LdapUtils.emptyLdapName(),
                getDefaultSearchControls(defaultSearchScope, RETURN_OBJ_FLAG, ALL_ATTRIBUTES),
                clazz);
    }
```
Wich results in that the base attribute in the @Entity doens't work for LdapRepositorys findAll() method.

**Comment by trorbyte** [comment]
Also seeing this on any find methods

**Comment by lewisheadden** [comment]
I am also seeing this. Has this really been open for over a year? It seems like a fairly major bug in Spring LDAP that should be more clearly documented if it can't be fixed more easily.

**Comment by bhuism** [comment]
@lewisheadden I've fixed this at https://github.com/bhuism/spring-ldap.

**Comment by hoalex** [comment]
I'm also observing this behaviour. Any chance to get this fixed?

**Comment by dustins** [comment]
#469

**Comment by dustins** [comment]
#470

Ok, if the API can't be changed for the foreseeable future, here is at least a fix so that the correct search is performed.

**Comment by jkeruzec** [comment]
Any idea when the pull request will be accepted ?

**Comment by barchiesi** [comment]
Will #470 ever get merged? If not, what would the recommended way of getting around the issue?

**Comment by jzheaux** [comment]
> Should `ObjectDirectoryMapper` override the `baseFilter` when calling `ObjectDirectoryMapper#filterFor`

@lewisheadden, I don't think so. Ultimately, `DirContext#search` keeps the concepts of `base` and `filter` separate and Spring LDAP should probably preserve that in its API.

> Or should we be checking for class annotations in https://github.com/spring-projects/spring-ldap/blob/master/core/src/main/java/org/springframework/ldap/core/LdapTemplate.java#L1814?

This could work, though it seems to me that this work is already done in `ObjectDirectoryMapper` and it would be nice to maintain the abstraction so it can be switched out.

I'm inclined to introduce something that allows `LdapTemplate` to derive the base from `ObjectDirectoryMapper`, though I still wonder the right way to do it. LDAP 2.4's baseline is JDK 8, so adding an interface method to `ObjectDirectoryMapper` is now an option.

**Comment by mrahman-vf-ic-bd** [comment]
Any idea when the pull request will be accepted ?
